### PR TITLE
Ports over Mortars and Pestles

### PR DESCRIPTION
**Pull request by Trilbyspaceclone**
## About The Pull Request

Ports over Mortars and Pestles from TG
Made with Wood and Metal allows you to grind up chems without needing power at the cost of time and staminda

## Why It's Good For The Game

Makes chemists no longer fight over as much the grinder. Helps ashies out a bit with getting some basic chem stuff going on miner slave and the like.
Getto chems without needing holodeck or stealing a grinder is amazing!

## Changelog
:cl:
add: Ports over TG's Mortars and Pestles.
/:cl:

ports - [ready]ghetto grinder #43600
ports - Mortar and Pestle Fixes #48086
Links -
https://github.com/tgstation/tgstation/pull/43600
https://github.com/tgstation/tgstation/pull/48086

### CODE REVIEW

**Review by Linzolle — APPROVED** [review]

File: code/modules/reagents/reagent_containers/glass.dm [hunk 1]
```diff
@@ -388,3 +388,65 @@
...
+				QDEL_NULL(grinded)
+				return
+			return
+		else
+			to_chat(user, "<span class='warning'>There is nothing to grind!</span>")
+			return
+	if(grinded)
+		to_chat(user, "<span class='warning'>There is something inside already!</span>")
+		return
+	if(I.juice_results || I.grind_results)
```
Comment (Linzolle) [inline code/modules/reagents/reagent_containers/glass.dm:450]: should this not also have `I.reagents`?

File: code/modules/reagents/reagent_containers/glass.dm [hunk 2]
```diff
@@ -388,3 +388,65 @@
...
+	if(grinded)
+		grinded.forceMove(drop_location())
+		grinded = null
+		to_chat(user, "<span class='notice'>You eject the item inside.</span>")
+
+/obj/item/reagent_containers/glass/mortar/attackby(obj/item/I, mob/living/carbon/human/user)
+	..()
+	if(istype(I,/obj/item/pestle))
+		if(grinded)
+			if(user.getStaminaLoss() > 50)
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm]:
```suggestion
			if(IS_STAMCRIT(user))
```

File: code/modules/reagents/reagent_containers/glass.dm (hunk 1 above)
Comment (Ghommie) [reply code/modules/reagents/reagent_containers/glass.dm:450]: Apparently this is just how it's on tgstation.

File: code/modules/reagents/reagent_containers/glass.dm [hunk 3]
```diff
@@ -388,3 +388,65 @@
...
+	desc = "A specially formed bowl of ancient design. It is possible to crush or juice items placed in it using a pestle; however the process, unlike modern methods, is slow and physically exhausting. Alt click to eject the item."
+	icon_state = "mortar"
+	amount_per_transfer_from_this = 10
+	possible_transfer_amounts = list(5, 10, 15, 20, 25, 30, 50)
+	item_flags = NO_MAT_REDEMPTION
+	reagent_flags = OPENCONTAINER
+	spillable = TRUE
+	var/obj/item/grinded
+
+/obj/item/reagent_containers/glass/mortar/AltClick(mob/user)
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm:412]:
```suggestion
/obj/item/reagent_containers/glass/mortar/AltClick(mob/user)
	. = ..()
```

File: code/modules/reagents/reagent_containers/glass.dm [hunk 4]
```diff
@@ -388,3 +388,65 @@
...
+	if(grinded)
+		grinded.forceMove(drop_location())
+		grinded = null
+		to_chat(user, "<span class='notice'>You eject the item inside.</span>")
```
Comment (Ghommie) [inline code/modules/reagents/reagent_containers/glass.dm:417]:
```suggestion
		to_chat(user, "<span class='notice'>You eject the item inside.</span>")
		return TRUE
```

**Review by Ghommie — CHANGES_REQUESTED** [review]

**Review by Ghommie — APPROVED** [review]

**Pull request merged**